UmbrellaPipeline/pathGeneration/tree.py

```diff
@@ -0,0 +1,189 @@
+                pos = []
+                for i in coordinates:
+                    pos.append(list(i.value_in_unit(i.unit)))
+                self.tree = KDTree(pos)
+            except AttributeError:
+                raise ValueError("no _unit provided.")
+        self.possibleNeighbours = [
+            [-1, -1, -1],
+            [-1, -1, 0],
+            [-1, -1, 1],
+            [-1, 0, -1],
+            [-1, 0, 0],
+            [-1, 0, 1],
+            [-1, 1, -1],
+            [-1, 1, 0],
+            [-1, 1, 1],
+            [0, -1, -1],
+            [0, -1, 0],
+            [0, -1, 1],
+            [0, 0, -1],
+            [0, 0, 1],
+            [0, 1, -1],
+            [0, 1, 0],
+            [0, 1, 1],
+            [1, -1, -1],
+            [1, -1, 0],
+            [1, -1, 1],
+            [1, 0, -1],
+            [1, 0, 0],
+            [1, 0, 1],
+            [1, 1, -1],
+            [1, 1, 0],
+            [1, 1, 1],
+        ]
+
+    @classmethod
+    def treeFromFiles(
+        cls, pdb: str or app.PDBFile, psf: str or app.CharmmPsfFile,
+    ):
+        """
+        Constructor for grid. takes in psf and pdb files generated from charmmgui and generates a grid where all points with a protein atom are true. every other gridpoint is False.
+
+        Args:
+            pdbfile (str): give either path to pdb file as string or an openmm.app.PDBFile object.
+            psffile (str): give either path to psf file as string or an openmm.app.CharmmPsfFile object.
+        Returns:
+            Grid: KDTree with the protein positions used for the adapted A* algorithm
+        """
+        try:
+            pdb = app.PDBFile(pdb)
+        except TypeError:
+            pdb = pdb
+        try:
+            psf = app.CharmmPsfFile(psf)
+        except TypeError:
+            psf = psf
+
+        indices = get_indices(psf.atom_list)
+        coords = []
+        unit = pdb.positions.unit
+        for i in indices:
+            coords.append(list(pdb.positions[i].value_in_unit(unit)))
+
+        return cls(_unit=unit, coordinates=coords)
+
+    def nodeFromFiles(self, psf: str, pdb: str, name: str) -> TreeNode:
+        """
+        calculates the centroid coordinates of the ligand and returns the grid node closest to the centriod Cordinates.
+
+        Args:
+            psf (str): give either path to pdb file as string or an openmm.app.PDBFile object.
+            pdb (str): give either path to psf file as string or an openmm.app.CharmmPsfFile object.
+            name (str): name of the residue that is the starting point.
+
+        Returns:
+            Node: grid node closest to ligand centroid.
+
+        TODO: add support for center of mass -> more complicated since it needs an initialized system.
+        """
+        try:
+            pdb = app.PDBFile(pdb)
+        except TypeError:
+            pdb = pdb
+        try:
+            psf = app.CharmmPsfFile(psf)
+        except TypeError:
+            psf = psf
+
```

Comment:
do you also have a check that raises an error if the file is neither a PDB nor a PSF?